### DISCUSSION TITLE
python + tkinter in dark mode

### DISCUSSION THREAD

**Discussion by m000** [Homebrew/discussions#247] [Everyday usage] [answered]
It seems that tkinter won't display colors properly when using dark mode. This can be demonstrated by running the simple [hello world][tkinter-hw] program from the tkinter module documentation. 

The bug has been reported to the Python bug tracker as [issue 42541][bpo42541], and traces back to the Tcl/Tk C libraries. This means that it is highly unlikely that it will be fixed on the Python side.

I can confirm that compiling against Homebrew-installed tcl-tk (currently v8.6.10) solves the issue, but adds a new dependency to Python.

The bug also affects [IDLE][idle]. It's not unusable, but configuration and other dialogs are ugly. Also, when linking against the system Tcl/Tk, you get a warning that you should upgrade:
```
WARNING: The version of Tcl/Tk (8.5.9) in use may be unstable. Visit
http://www.python.org/download/mac/tcltk/ for current information.
```

So maybe it's time to do just that?

[bpo42541]: https://bugs.python.org/issue42541
[tkinter-hw]: https://docs.python.org/3/library/tkinter.html#a-simple-hello-world-program
[idle]: https://docs.python.org/3/library/idle.html

**Comment by ushuz** [comment, marked as answer]
Should be fixed by https://github.com/Homebrew/homebrew-core/pull/67378

**Reply by m000** [reply]
Glad that someone took notice :)
Proposing to link with homebrew's packages is typically not well received, so I wouldn't like to be the one to raise the issue first.

**Reply by carlocab** [reply]
Fix has been merged, FYI
